### PR TITLE
Validator of parameters

### PR DESCRIPTION
**Pull request by ManuelFernando**

### CODE REVIEW

File: test/unit/parameters.test.js [hunk 1]
```diff
@@ -0,0 +1,340 @@
+/* global describe, it */
+
+import assert from 'assert';
+import PubNub from '../../src/node/index';
+
+let pubnub = new PubNub({ subscribeKey: 'mySubKey', publishKey: 'myPublishKey', uuid: 'myUUID', authKey: 'myAuthKey' });
```
Comment (MaxPresman) [inline test/unit/parameters.test.js]: @ManuelFernando can we move those tests into the specific test files for the endpoints?

File: src/core/parameters.js [hunk 2]
```diff
@@ -0,0 +1,175 @@
+const endpoints = {
+  setup: {
```
Comment (MaxPresman) [inline src/core/parameters.js]: @ManuelFernando can we move those configs into individual files? this way when we modify the params, we can easily adjust.

File: test/unit/parameters.test.js (hunk 1 above)
Comment (ManuelFernando) [inline test/unit/parameters.test.js]: sure

File: src/core/parameters.js (hunk 2 above)
Comment (ManuelFernando) [inline src/core/parameters.js]: sure

**Comment by ManuelFernando** [comment]
@MaxPresman changes ready!

File: src/core/parameters/add_channels.js [hunk 3]
```diff
@@ -0,0 +1,6 @@
+/* @flow */
+export default {
```
Comment (MaxPresman) [inline src/core/parameters/add_channels.js]: would be great to have the parameters code inside the endpoint itself.

Comment (ManuelFernando) [inline src/core/parameters/add_channels.js]: that is easy to move, but for the case of subscribe, set_state and unsubscribe are managed from the subscription manager, then can we move these config to subscription manager?

**Comment by ManuelFernando** [comment]
@MaxPresman ready :)

**Pull request closed without merging**